Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp

```diff
@@ -0,0 +1,244 @@
+#include "Carla.h"
+#include "Engine/World.h"
+#include "Game/CarlaGameModeBase.h"
+#include "Settings/CarlaSettings.h"
+#include "CarlaSettingsDelegate.h"
+
+UCarlaSettingsDelegate::UCarlaSettingsDelegate() :
+  ActorSpawnedDelegate(FOnActorSpawned::FDelegate::CreateUObject(this, &UCarlaSettingsDelegate::OnActorSpawned))
+{
+}
+
+void UCarlaSettingsDelegate::RegisterSpawnHandler(UWorld *InWorld)
+{
+  check(InWorld!=nullptr);
+  auto GameInstance  = Cast<UCarlaGameInstance>(InWorld->GetGameInstance());
+  check(GameInstance!=nullptr);
+  CarlaSettings = &GameInstance->GetCarlaSettings();
+  check(CarlaSettings!=nullptr);
+  InWorld->AddOnActorSpawnedHandler(ActorSpawnedDelegate);
+}
+
+void UCarlaSettingsDelegate::OnActorSpawned(AActor* InActor)
+{
+  check(CarlaSettings!=nullptr);
+  if (InActor != nullptr && IsValid(InActor) && !InActor->IsPendingKill()) 
+  {
+     //apply settings for this actor for the current quality level
+	   TArray<UActorComponent*> components = InActor->GetComponentsByClass(UPrimitiveComponent::StaticClass());
+	   float dist = CarlaSettings->LowStaticMeshMaxDrawDistance;
+     const float maxscale = InActor->GetActorScale().GetMax();
+     #define _MAX_SCALE_SIZE 50.0f
+     
+	   if(maxscale>_MAX_SCALE_SIZE) {
+        dist *= 100.0f;
+     }
+
+	   for(int32 i=0; i<components.Num(); i++)
+  	 {
+		  UPrimitiveComponent* primitivecomponent = Cast<UPrimitiveComponent>(components[i]);
+		  if(IsValid(primitivecomponent))
+		  {
+			 primitivecomponent->SetCullDistance(dist);
+		  }
+	   }
+  }
+}
+
+
+void UCarlaSettingsDelegate::ApplyQualitySettingsLevelPostRestart() const
+{
+	switch(CarlaSettings->GetQualitySettingsLevel())
+	{
+	  case EQualitySettingsLevel::Low: 
+	  	ApplyLowQualitySettings();
+		  break;
+	  case EQualitySettingsLevel::Medium: 
+      /** @todo */
```

Comment:
Maybe we should log an error here warning that it has no effect

Reply:
You are right.